src/main/java/domain/Cars.java

```diff
@@ -1,35 +1,70 @@
 package domain;
 
+import static domain.RandomNumber.generateRandomOutOfTen;
+import static domain.RandomNumber.isOverFourOutOfTen;
+
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class Cars {
 
     private final List<Car> cars;
 
-    public Cars(UserInput userInput) {
-        int numbersOfCar = userInput.getNumbersOfCar();
-        List<Car> cars = createCars(numbersOfCar);
+    public Cars(List<Car> cars) {
         this.cars = cars;
     }
 
-    private static List<Car> createCars(int numbersOfCar) {
+    public List<Car> createCars(String[] namesForCars, int numbersOfCar) {
         List<Car> cars = new ArrayList<>();
         for (int i = 0; i < numbersOfCar; i++) {
-            cars.add(new Car());
+            cars.add(new Car(namesForCars[i]));
         }
         return cars;
     }
 
     public void makeCarsMove() {
-        cars.forEach(Car::attemptMove);
+        for (int i = 0; i < cars.size(); i++) {
+            cars.get(i).attemptMove(isMoved());
+        }
+    }
+
+    private boolean isMoved() {
+        return isOverFourOutOfTen(generateRandomOutOfTen());
     }
 
-    public int getSizeOfCars() {
+    public int countCars() {
         return cars.size();
     }
 
-    public int getDistance(int i) {
-        return cars.get(i).getCarCurrentDistance();
+    public Car getEachCar(int carIndex) {
+        return cars.get(carIndex);
+    }
+
+
+    public String[] getWinnerNames() {
+        List<Car> winners = checkCarWithFarthestDistance();
+        return winners.stream()
+                .map(Car::getCarName)
+                .toArray(String[]::new);
+    }
+
+    private List<Car> checkCarWithFarthestDistance() {
+        int farthestDistance = getFarthestDistance();
+        return getCarsWithFarthestDistance(farthestDistance);
+    }
+
+    private int getFarthestDistance() {
+        int farthestDistance = 0;
+        for (Car car : this.cars) {
+            farthestDistance = Math.max(farthestDistance, car.getCarCurrentDistance());
+        }
+        return farthestDistance;
+    }
```

Comment:
```suggestion
    private int getFarthestDistance() {
        return cars.stream()
                .mapToInt(Car::getCarCurrentDistance)
                .max()
                .orElse(0);
    }
```

요렇게 stream 을 이용해줄 수 도 있겠네요 😉